authz/grpc_audit_end2end_test.go

```diff
@@ -0,0 +1,304 @@
+package authz
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/grpc/credentials"
+	testgrpc "google.golang.org/grpc/interop/grpc_testing"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/testdata"
+
+	_ "google.golang.org/grpc/authz/audit/stdout"
+)
+
+type testServer struct {
+	testgrpc.UnimplementedTestServiceServer
+}
+
+func (s *testServer) UnaryCall(ctx context.Context, req *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	return &testpb.SimpleResponse{}, nil
+}
+
+type statAuditLogger struct {
+	Stat         map[bool]int
+	stdoutLogger audit.Logger
```

Comment:
If we are not doing anything with it, why are we including the stdout logger?

Reply:
I have the following 2 thoughts 

1. Making sure that `stdoutLogger` can be referenced from outside it's package and usable externally (unit tests don't cover it)
2. It execution doesn't fail while external usage (details of execution are covered by unit tests). 

So for a second theoretical xDS logger I'd also like to include it here

Reply:
In that case, can we simply add `stdout_logger` as another logger in the policy so that we are also testing multiple loggers. Because embedding a logger is the implementation detail of a custom logger that we don't really need to cover in our tests.